### PR TITLE
MINT-421 Changes in Vote contracts

### PR DESCRIPTION
**Pull request by alesanro**
- fix event generate (`PollCreated`)
- add update details events
- add test-cases for this issue

## Note

Details (title + description + other info), options - all these fields in case of long text should be stored in IPFS and should take only hash data (bytes32)

### CODE REVIEW

File: contracts/vote/PollManager.sol [hunk 1]
```diff
@@ -148,6 +158,70 @@ contract PollManager is PollEmitter, Vote {
...
+        if (store.includes(optionsId, bytes32(_pollId), _option)) {
+            return _emitError(ERROR_VOTE_INVALID_PARAMETER);
+        }
+
+        store.add(optionsId, bytes32(_pollId), _option);
+        _emitOptionAdded(_pollId, _option, store.count(optionsId, bytes32(_pollId)));
+        return OK;
+    }
+
+    function removePollOption(uint _pollId, bytes32 _option) returns (uint errorCode) {
```
Comment (ahiatsevich) [inline contracts/vote/PollManager.sol]: TODO: check access rights?

File: contracts/vote/PollManager.sol [hunk 2]
```diff
@@ -148,6 +158,70 @@ contract PollManager is PollEmitter, Vote {
...
+
+        if (store.get(detailsIpfsHash, _pollId) != _detailsIpfsHash) {
+            store.set(detailsIpfsHash, _pollId, _detailsIpfsHash);
+        }
+
+        _emitPollDetailsUpdated(_pollId);
+        return OK;
+    }
+
+    function addPollOption(uint _pollId, bytes32 _option) returns (uint errorCode) {
```
Comment (ahiatsevich) [inline contracts/vote/PollManager.sol]: TODO: check access rights?

File: contracts/vote/PollManager.sol [hunk 3]
```diff
@@ -148,6 +158,70 @@ contract PollManager is PollEmitter, Vote {
         errorCode = _checkAndEmitError(result);
     }
 
+    function updatePollDetailsIpfsHash(uint _pollId, bytes32 _detailsIpfsHash) returns (uint errorCode) {
```
Comment (ahiatsevich) [inline contracts/vote/PollManager.sol]: TODO: check access rights?

File: contracts/vote/PollManager.sol (hunk 1 above)
Comment (alesanro) [inline contracts/vote/PollManager.sol]: It is on the next line, validation is performed on poll's onwnership

File: contracts/vote/PollManager.sol (hunk 2 above)
Comment (alesanro) [inline contracts/vote/PollManager.sol]: It is on the next line, validation is performed on poll's ownership

File: contracts/vote/PollDetails.sol [hunk 4]
```diff
@@ -63,46 +69,54 @@ contract PollDetails is Vote {
         }
     }
 
-    function getMemberVotesForPoll(uint _id) constant returns (uint result) {
+    function getMemberVotesForPoll(uint _id) onlyExistedPoll(_id) constant returns (uint result) {
```
Comment (ahiatsevich) [inline contracts/vote/PollDetails.sol]: TODO: is 0 a valid result for `getMemberVotesForPoll` if poll exists?

Comment (ahiatsevich) [inline contracts/vote/PollDetails.sol]: `function getMemberVotesForPoll(uint _id) onlyExistedPoll(_id) constant returns (uint memberVotes, uint resultCode)`

Comment (ahiatsevich) [inline contracts/vote/PollDetails.sol]: 0-resultCode will be handled as Unauthorized, not as PollDoesNotExist

File: contracts/vote/PollManager.sol [hunk 5]
```diff
@@ -31,17 +35,15 @@ contract PollManager is PollEmitter, Vote {
         return TimeHolder(timeHolder).totalSupply() / 10000 * store.get(sharesPercent);
     }
 
-    function NewPoll(bytes32[16] _options, bytes32[4] _ipfsHashes, bytes32 _title, bytes32 _description, uint _votelimit, uint _deadline) returns (uint errorCode) {
+    function NewPoll(bytes32[16] _options, bytes32[4] _ipfsHashes, bytes32 _detailsIpfsHash, uint _votelimit, uint _deadline) returns (uint errorCode) {
```
Comment (ahiatsevich) [inline contracts/vote/PollManager.sol]: TODO: input params sanity check?

**Pull request merged**